classy-prelude/src/ClassyPrelude.hs

```diff
@@ -646,3 +655,56 @@ hPut h = liftIO . Data.ByteString.hPut h
 -- @since 1.2.0
 hGetChunk :: MonadIO m => Handle -> m ByteString
 hGetChunk = liftIO . flip Data.ByteString.hGetSome defaultChunkSize
+
+-- | Write a character to stdout
+--
+-- Uses system locale settings
+putChar :: MonadIO m => Char -> m ()
+putChar = liftIO . Prelude.putChar
+
+-- | Write a Text to stdout
+--
+-- Uses system locale settings
+--
+-- @since 1.3.1
+putStr :: MonadIO m => Text -> m ()
+putStr = liftIO . TextIO.putStr
+
+-- | Write a Text followed by a newline to stdout
+--
+-- Uses system locale settings
+--
+-- @since 1.3.1
+putStrLn :: MonadIO m => Text -> m ()
+putStrLn = liftIO . TextIO.putStrLn
+
+-- | Read a character from stdin
+--
+-- Uses system locale settings
+--
+-- @since 1.3.1
+getChar :: MonadIO m => m Char
+getChar = liftIO Prelude.getChar
+
+-- | Read a line from stdin
+--
+-- Uses system locale settings
+getLine :: MonadIO m => m Text
+getLine = liftIO TextIO.getLine
+
+-- | Read all input from stdin into a Text
+--
+-- Uses system locale settings
+--
+-- @since 1.3.1
+getContents :: MonadIO m => m Text
+getContents = liftIO TextIO.getContents
+
+-- | Takes a function of type 'Text -> Text' and passes all input on stdin
+-- to it, then prints result to stdout
+--
+-- Uses system locale settings
+--
+-- @since 1.3.1
+interact :: MonadIO m => (Text -> Text) -> m ()
```

Comment:
Is a strict `Text` the correct data type to use here? I'm generally opposed to lazy I/O, but a strict I/O version of `interact` is pretty confusing. At the very least, it should be documented. (Same basic idea applies to `getContents`, though due to the type signature of `interact` the downsides of lazy I/O are much less severe.)

Reply:
Switched it to lazy IO. It does have the downside here that `LText`s have to be used instead of `Text`s, while in the Prelude, it's `String`s either way.

Reply:
I agree that it's a downside.